File: lib/src/Models/quiz/quiz.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:hive/hive.dart';

import '../question/question.dart';
import '../quiz_result.dart';

part 'quiz.freezed.dart';
part 'quiz.g.dart';

@unfreezed
class Quiz with _$Quiz{
  @HiveType(typeId: 1, adapterName: 'QuizAdapter')
  factory Quiz({
    @HiveField(0)
    required final int quizId,
    @HiveField(1)
    required final String title,
    @HiveField(2)
    final String? description,
    @Default([])
    @HiveField(3)
    List<Question> questions,
    @Default(false)
    @HiveField(4)
    bool finished,
    @HiveField(5)
    QuizResult? result
  }) = _Quiz;

  Quiz._();

  factory Quiz.fromJson(Map<String, Object?> json) => _$QuizFromJson(json);
}

File: api/PhilosophyLearnAppAPI/Models/Quiz.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PhilosophyLearnAppAPI.Models
{
    public class Quiz
    {
        [Key]
        public int QuizId { get; set; }

        [Required]
        [MaxLength(50)]
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public bool Finished { get; set; } = false;

        [ForeignKey("Topic")]
        public int TopicId { get; set; }

        public virtual ICollection<Question> Questions { get; set; } = [];
    }
}
